jupyterhub/orm.py

```diff
@@ -222,9 +267,22 @@ def check_routes(self, user_dict, routes=None):
                 # catch filter bug, either in sqlalchemy or my understanding of its behavior
```

Comment:
This hasn't changed, it was introduced some time ago. I still don't understand how it was happening, but I've never seen it personally.